### DISCUSSION TITLE
How I can set resource label in only first letter case?

### DISCUSSION THREAD

**Discussion by PetroZaburko** [filamentphp/filament#10384] [Help] [answered]
### Package
filament/filament
### Package Version
3.0.94 
### Laravel Version
10.31.0
### Livewire Version
 v3.1.0
### PHP Version
8.2.2
### Question

How I can set in resource single and plural model label in only "first letter upper" case?

 I used methods:

`    public static function getModelLabel(): string
    {
        return __('app.post.single');
    }`
and
`    public static function getPluralModelLabel(): string
    {
        return __('app.post.plural');
    }`

But if I used label with two or more words (in my case 'Knowledge base') I had result in "kebab" case, like: 
![Screenshot from 2023-12-19 14-44-12](https://github.com/filamentphp/filament/assets/56628382/4eea7bea-a04c-4022-9cf3-276a7654386d)

**Comment by leandrocfe** [comment, marked as answer]
You can use `getTitle()` in each resource page, ListPage.php for example

```php
public function getTitle(): string|Htmlable
{
    return __('app.post.plural');
}
```

If you want to customize the menu item, use `getNavigationLabel()` in the resource

```php
public static function getNavigationLabel(): string
{
    return __('app.post.plural');
}
```

**Reply by PetroZaburko** [reply]
Thanks, it helps! I don't knew about  `getTitle()` method in each resource page.

**Reply by leandrocfe** [reply]
docs: https://filamentphp.com/docs/3.x/panels/pages#customizing-the-page-title

**Comment by buderdene** [comment]
File map : Vendor/Filament/Filament/scr/Resource/Resource/php


public static function getTitleCasePluralModelLabel(): string
    {
        if (! static::hasTitleCaseModelLabel()) {
            return static::getPluralModelLabel();
        }
        return static::getPluralModelLabel();
        //return Str::ucwords(static::getPluralModelLabel());
    }

**Reply by micraux** [reply]
Not recommended
